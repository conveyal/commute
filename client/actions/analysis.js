import moment from 'moment'
import {push} from 'react-router-redux'
import {createAction} from 'redux-actions'
import uuid from 'uuid'

const addAnalysis = createAction('add analysis')
export const createAnalysis = (newAnalysis, organizationId) => {
  newAnalysis.id = uuid.v4()
  newAnalysis.lastRunDateTime = moment().unix()
  const commuters = newAnalysis.commuters
  delete newAnalysis.commuters
  return [
    addAnalysis({ analysis: newAnalysis, organizationId }),
    push(`/organizations/${organizationId}/analysis/${newAnalysis.id}`),
    makeMockTrips(newAnalysis.id, organizationId, commuters)
  ] // TODO save to server
}

const deleteLocally = createAction('delete analysis')
/* const deleteOnServer = (id) =>
  serverAction({
    url: `/api/analysis/${id}`,
    options: {
      method: 'delete'
    }
  }) */ // TODO delete on server
export const deleteAnalysis = (id, organizationId) => [
  deleteLocally({ analysisId: id, organizationId }),
  push(`/organizations/${organizationId}`)
]

const makeMockTrips = (analysisId, organizationId, commuters) => {
  const trips = []
  const modes = ['bike', 'car', 'transit', 'walk']
  const randRange = () => Math.random() * 0.4 + 0.8
  commuters.forEach((commuter) => {
    const curTrip = {
      commuterId: commuter.id
    }
    const tripMetricBase = Math.random()
    let bestValue = 9999999
    let bestMode
    modes.forEach((mode) => {
      // generate random trips
      const modePctMultiplier = randRange()
      const cost = tripMetricBase * 34.7 * modePctMultiplier * randRange()
      const distance = tripMetricBase * 69 * modePctMultiplier * randRange()
      const time = tripMetricBase * 7500 * modePctMultiplier * randRange()
      curTrip[mode] = {cost, distance, time}
      const tripValue = cost + time * 0.01
      if (tripValue < bestValue) bestMode = mode
    })
    curTrip.mostLikely = Object.assign({ mode: bestMode }, curTrip[bestMode])
    trips.push(curTrip)
  })
  return sendMockTrips({analysisId, organizationId, trips})
}
const sendMockTrips = createAction('receive mock calculated trips')
