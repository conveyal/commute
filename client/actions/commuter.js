import {push} from 'react-router-redux'
import {createAction} from 'redux-actions'
import uuid from 'uuid'

const addCommuter = createAction('add commuter')
export const createCommuter = ({commuter, groupId, organizationId}) => {
  commuter.id = uuid.v4()
  return [
    addCommuter({commuter, groupId, organizationId}),
    push(`/organizations/${organizationId}/groups/${groupId}`)
  ] // TODO save to server
}

const deleteLocally = createAction('delete commuter')
/* const deleteOnServer = (id) =>
  serverAction({
    url: `/api/commuter/${id}`,
    options: {
      method: 'delete'
    }
  }) */ // TODO delete on server
export const deleteCommuter = ({commuterId, groupId, organizationId}) => [
  deleteLocally({commuterId, groupId, organizationId}),
  push(`/organizations/${organizationId}/groups/${groupId}`)
]

const updateLocally = createAction('update commuter')
/* const updateOnServer = (id) =>
  serverAction({
    url: `/api/commuter/${id}`,
    options: {
      method: 'update'
    }
  }) */ // TODO update on server
export const updateCommuter = ({commuter, groupId, organizationId}) => [
  updateLocally({commuter, groupId, organizationId}),
  push(`/organizations/${organizationId}/groups/${groupId}`)
]
