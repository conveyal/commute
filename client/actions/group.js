import {push} from 'react-router-redux'
import {createAction} from 'redux-actions'
import uuid from 'uuid'

const addGroup = createAction('add group')
const appendCommuters = createAction('append commuters')
export const appendToGroup = (newCommuters, groupId, organizationId) => {
  return [
    appendCommuters({ commuters: newCommuters, groupId, organizationId }),
    push(`/organizations/${organizationId}/groups/${groupId}`)
  ] // TODO save to server
}
export const createGroup = (newGroup, organizationId) => {
  newGroup.id = uuid.v4()
  newGroup.allAddressesGeocoded = false
  return [
    addGroup({ group: newGroup, organizationId }),
    push(`/organizations/${organizationId}/groups/${newGroup.id}`)
  ] // TODO save to server
}

const deleteLocally = createAction('delete group')
/* const deleteOnServer = (id) =>
  serverAction({
    url: `/api/group/${id}`,
    options: {
      method: 'delete'
    }
  }) */ // TODO delete on server
export const deleteGroup = (id, organizationId) => [
  deleteLocally({ groupId: id, organizationId }),
  push(`/organizations/${organizationId}`)
]
