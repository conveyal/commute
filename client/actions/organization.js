import {push} from 'react-router-redux'
import {createAction} from 'redux-actions'
import uuid from 'uuid'

const addLocally = createAction('add organization')
export const createOrganization = (newOrganization) => {
  newOrganization.id = uuid.v4()
  newOrganization.analyses = []
  newOrganization.analysesById = {}
  newOrganization.sites = []
  newOrganization.sitesById = {}
  newOrganization.groups = []
  newOrganization.groupsById = {}
  return [
    addLocally(newOrganization),
    push(`/organizations/${newOrganization.id}`)
  ] // TODO save to server
}

const deleteLocally = createAction('delete organization')
/* const deleteOnServer = (id) =>
  serverAction({
    url: `/api/organization/${id}`,
    options: {
      method: 'delete'
    }
  }) */ // TODO delete on server
export const deleteOrganization = (id) => [deleteLocally(id), push('/')]

export const updateOrganization = (organization) => [
  setLocally(organization),
  push(`/organizations/${organization.id}`)
]
/* const updateOnServer = (organization) =>
  serverAction({
    url: `/api/organization/${organization.id}`,
    options: {
      body: JSON.stringify(organization),
      method: 'put'
    }
  }) */  // TODO: update on server

export const setLocally = createAction('set organization')
