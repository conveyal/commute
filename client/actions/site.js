import {push} from 'react-router-redux'
import {createAction} from 'redux-actions'
import uuid from 'uuid'

const addSite = createAction('add site')
export const createSite = (newSite, organizationId) => {
  newSite.id = uuid.v4()
  return [
    addSite({ organizationId, site: newSite }),
    push(`/organizations/${organizationId}`)
  ] // TODO save to server
}

const deleteLocally = createAction('delete site')
/* const deleteOnServer = (id) =>
  serverAction({
    url: `/api/site/${id}`,
    options: {
      method: 'delete'
    }
  }) */ // TODO delete on server
export const deleteSite = (id, organizationId) => [
  deleteLocally({ siteId: id, organizationId }),
  push(`/organizations/${organizationId}`)
]

const updateLocally = createAction('update site')
/* const updateOnServer = (id) =>
  serverAction({
    url: `/api/site/${id}`,
    options: {
      method: 'update'
    }
  }) */ // TODO update on server
export const updateSite = (site, organizationId) => [
  updateLocally({ organizationId, site }),
  push(`/organizations/${organizationId}`)
]
