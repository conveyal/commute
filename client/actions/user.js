import {createAction} from 'redux-actions'

// login / logout
export const login = createAction('log in')
export const logout = createAction('log out')
