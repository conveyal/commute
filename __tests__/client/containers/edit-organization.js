/* global describe, expect, it */

import {mount} from 'enzyme'
import {mountToJson} from 'enzyme-to-json'
import React from 'react'
import {Provider} from 'react-redux'

import {expectCreateAction, expectDeleteOrganization} from '../../test-utils/actions'
import {makeMockStore, mockStores} from '../../test-utils/mock-data.js'

import EditOrganization from '../../../client/containers/edit-organization'

describe('Container > EditOrganization', () => {
  it('Create/Edit Organization View loads (create or edit mode)', () => {
    const mockStore = makeMockStore(mockStores.init)

    // mount component
    const tree = mount(
      <Provider store={mockStore}>
        <EditOrganization />
      </Provider>
    )
    expect(mountToJson(tree.find(EditOrganization))).toMatchSnapshot()
  })

  it('Create/Edit Organization View loads in edit mode', () => {
    const mockStore = makeMockStore(mockStores.oneSimpleOrganization)

    // mount component
    const tree = mount(
      <Provider store={mockStore}>
        <EditOrganization
          params={{organizationId: '1'}}
          />
      </Provider>
    )
    expect(mountToJson(tree.find(EditOrganization))).toMatchSnapshot()
  })

  it('Create organization', () => {
    const mockStore = makeMockStore(mockStores.init)

    // mount component
    const tree = mount(
      <Provider store={mockStore}>
        <EditOrganization />
      </Provider>
    )

    // give each text field some input
    tree.find('input').map((input) => input.simulate('change', {target: {value: 'My new value'}}))

    // submit form
    tree.find('form').find('button').simulate('click')

    // expect create action
    expectCreateAction(mockStore.getActions())
  })

  it('Update organization', () => {
    const mockStore = makeMockStore(mockStores.oneSimpleOrganization)

    // mount component
    const tree = mount(
      <Provider store={mockStore}>
        <EditOrganization
          params={{organizationId: '1'}}
          />
      </Provider>
    )

    // give each text field some input
    tree.find('input').map((input) => input.simulate('change', {target: {value: 'My new value'}}))

    // submit form
    tree.find('form').find('button').first().simulate('click')

    // expect update action
    const actions = mockStore.getActions()
    expect(actions.length).toBe(2)
    expect(actions).toMatchSnapshot()
  })

  it('Delete organization', () => {
    const mockStore = makeMockStore(mockStores.oneSimpleOrganization)
    window.confirm = () => true

    // Given a logged-in user is viewing the Create/Edit Organization View
    // mount component
    const tree = mount(
      <Provider store={mockStore}>
        <EditOrganization
          params={{organizationId: '1'}}
          />
      </Provider>
    )

    // When the user clicks the delete button for an existing organization
    // And the user confirms the Confirm Deletion dialog
    const deleteButton = tree.find('button').last()
    deleteButton.simulate('click')

    expectDeleteOrganization(mockStore.getActions())
  })
})
