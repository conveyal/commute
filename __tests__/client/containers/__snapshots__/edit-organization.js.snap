exports[`Container > EditOrganization Create organization 1`] = `
Object {
  "payload": Object {
    "analyses": Array [],
    "analysesById": Object {},
    "contact": "My new value",
    "email": "My new value",
    "groups": Array [],
    "groupsById": Object {},
    "logo_url": "My new value",
    "main_url": "My new value",
    "name": "My new value",
    "sites": Array [],
    "sitesById": Object {},
  },
  "type": "add organization",
}
`;

exports[`Container > EditOrganization Create organization 2`] = `
Object {
  "payload": Object {
    "args": Array [
      "/organizations/new-entity-id",
    ],
    "method": "push",
  },
  "type": "@@router/CALL_HISTORY_METHOD",
}
`;

exports[`Container > EditOrganization Create/Edit Organization View loads (create or edit mode) 1`] = `
<Connect(EditOrganization)>
  <EditOrganization
    create={[Function]}
    delete={[Function]}
    update={[Function]}>
    <Grid
      bsClass="container"
      componentClass="div"
      fluid={false}>
      <div
        className="container">
        <Row
          bsClass="row"
          componentClass="div">
          <div
            className="row">
            <Col
              bsClass="col"
              componentClass="div"
              xs={12}>
              <div
                className="col-xs-12">
                <h3>
                  <span>
                    Create Organization
                  </span>
                  <Button
                    active={false}
                    block={false}
                    bsClass="btn"
                    bsStyle="default"
                    className="pull-right"
                    disabled={false}>
                    <button
                      className="pull-right btn btn-default"
                      disabled={false}
                      type="button">
                      <Link
                        onlyActiveOnIndex={false}
                        style={Object {}}
                        to="/">
                        <a
                          onClick={[Function]}
                          style={Object {}}>
                          <Icon
                            type="arrow-left">
                            <i
                              className="fa fa-arrow-left fa-fw " />
                          </Icon>
                        </a>
                      </Link>
                    </button>
                  </Button>
                </h3>
                <form>
                  <FieldGroup
                    label="Name"
                    name="name"
                    onChange={[Function]}
                    placeholder="Enter name"
                    type="text"
                    value={undefined}>
                    <FormGroup
                      bsClass="form-group"
                      controlId="group-item-name">
                      <div
                        className="form-group">
                        <ControlLabel
                          bsClass="control-label"
                          srOnly={false}>
                          <label
                            className="control-label"
                            htmlFor="group-item-name">
                            Name
                          </label>
                        </ControlLabel>
                        <FormControl
                          bsClass="form-control"
                          componentClass="input"
                          onChange={[Function]}
                          placeholder="Enter name"
                          type="text"
                          value="">
                          <input
                            className="form-control"
                            id="group-item-name"
                            onChange={[Function]}
                            placeholder="Enter name"
                            type="text"
                            value="" />
                        </FormControl>
                      </div>
                    </FormGroup>
                  </FieldGroup>
                  <FieldGroup
                    label="Main URL"
                    name="main_url"
                    onChange={[Function]}
                    placeholder="Enter url"
                    type="text"
                    value={undefined}>
                    <FormGroup
                      bsClass="form-group"
                      controlId="group-item-main_url">
                      <div
                        className="form-group">
                        <ControlLabel
                          bsClass="control-label"
                          srOnly={false}>
                          <label
                            className="control-label"
                            htmlFor="group-item-main_url">
                            Main URL
                          </label>
                        </ControlLabel>
                        <FormControl
                          bsClass="form-control"
                          componentClass="input"
                          onChange={[Function]}
                          placeholder="Enter url"
                          type="text"
                          value="">
                          <input
                            className="form-control"
                            id="group-item-main_url"
                            onChange={[Function]}
                            placeholder="Enter url"
                            type="text"
                            value="" />
                        </FormControl>
                      </div>
                    </FormGroup>
                  </FieldGroup>
                  <FieldGroup
                    label="Logo URL"
                    name="logo_url"
                    onChange={[Function]}
                    placeholder="Enter logo url"
                    type="text"
                    value={undefined}>
                    <FormGroup
                      bsClass="form-group"
                      controlId="group-item-logo_url">
                      <div
                        className="form-group">
                        <ControlLabel
                          bsClass="control-label"
                          srOnly={false}>
                          <label
                            className="control-label"
                            htmlFor="group-item-logo_url">
                            Logo URL
                          </label>
                        </ControlLabel>
                        <FormControl
                          bsClass="form-control"
                          componentClass="input"
                          onChange={[Function]}
                          placeholder="Enter logo url"
                          type="text"
                          value="">
                          <input
                            className="form-control"
                            id="group-item-logo_url"
                            onChange={[Function]}
                            placeholder="Enter logo url"
                            type="text"
                            value="" />
                        </FormControl>
                      </div>
                    </FormGroup>
                  </FieldGroup>
                  <FieldGroup
                    label="Contact"
                    name="contact"
                    onChange={[Function]}
                    placeholder="Enter contact"
                    type="text"
                    value={undefined}>
                    <FormGroup
                      bsClass="form-group"
                      controlId="group-item-contact">
                      <div
                        className="form-group">
                        <ControlLabel
                          bsClass="control-label"
                          srOnly={false}>
                          <label
                            className="control-label"
                            htmlFor="group-item-contact">
                            Contact
                          </label>
                        </ControlLabel>
                        <FormControl
                          bsClass="form-control"
                          componentClass="input"
                          onChange={[Function]}
                          placeholder="Enter contact"
                          type="text"
                          value="">
                          <input
                            className="form-control"
                            id="group-item-contact"
                            onChange={[Function]}
                            placeholder="Enter contact"
                            type="text"
                            value="" />
                        </FormControl>
                      </div>
                    </FormGroup>
                  </FieldGroup>
                  <FieldGroup
                    label="Email"
                    name="email"
                    onChange={[Function]}
                    placeholder="Enter email"
                    type="text"
                    value={undefined}>
                    <FormGroup
                      bsClass="form-group"
                      controlId="group-item-email">
                      <div
                        className="form-group">
                        <ControlLabel
                          bsClass="control-label"
                          srOnly={false}>
                          <label
                            className="control-label"
                            htmlFor="group-item-email">
                            Email
                          </label>
                        </ControlLabel>
                        <FormControl
                          bsClass="form-control"
                          componentClass="input"
                          onChange={[Function]}
                          placeholder="Enter email"
                          type="text"
                          value="">
                          <input
                            className="form-control"
                            id="group-item-email"
                            onChange={[Function]}
                            placeholder="Enter email"
                            type="text"
                            value="" />
                        </FormControl>
                      </div>
                    </FormGroup>
                  </FieldGroup>
                  <Button
                    active={false}
                    block={false}
                    bsClass="btn"
                    bsStyle="success"
                    disabled={false}
                    onClick={[Function]}>
                    <button
                      className="btn btn-success"
                      disabled={false}
                      onClick={[Function]}
                      type="button">
                      Create
                    </button>
                  </Button>
                </form>
              </div>
            </Col>
          </div>
        </Row>
      </div>
    </Grid>
  </EditOrganization>
</Connect(EditOrganization)>
`;

exports[`Container > EditOrganization Create/Edit Organization View loads in edit mode 1`] = `
<Connect(EditOrganization)
  params={
    Object {
      "organizationId": "1",
    }
  }>
  <EditOrganization
    create={[Function]}
    delete={[Function]}
    editMode={true}
    organization={
      Object {
        "analyses": Array [],
        "analysesById": Object {},
        "contact": "someone",
        "email": "abc@def.ghi",
        "groups": Array [],
        "groupsById": Object {},
        "id": "1",
        "logo_url": "https://placekitten.com/80/80",
        "main_url": "https://placekitten.com/",
        "name": "Mock Organization",
        "sites": Array [],
        "sitesById": Object {},
      }
    }
    params={
      Object {
        "organizationId": "1",
      }
    }
    update={[Function]}>
    <Grid
      bsClass="container"
      componentClass="div"
      fluid={false}>
      <div
        className="container">
        <Row
          bsClass="row"
          componentClass="div">
          <div
            className="row">
            <Col
              bsClass="col"
              componentClass="div"
              xs={12}>
              <div
                className="col-xs-12">
                <h3>
                  <span>
                    Edit Organization
                  </span>
                  <Button
                    active={false}
                    block={false}
                    bsClass="btn"
                    bsStyle="default"
                    className="pull-right"
                    disabled={false}>
                    <button
                      className="pull-right btn btn-default"
                      disabled={false}
                      type="button">
                      <Link
                        onlyActiveOnIndex={false}
                        style={Object {}}
                        to="/">
                        <a
                          onClick={[Function]}
                          style={Object {}}>
                          <Icon
                            type="arrow-left">
                            <i
                              className="fa fa-arrow-left fa-fw " />
                          </Icon>
                        </a>
                      </Link>
                    </button>
                  </Button>
                </h3>
                <form>
                  <FieldGroup
                    label="Name"
                    name="name"
                    onChange={[Function]}
                    placeholder="Enter name"
                    type="text"
                    value="Mock Organization">
                    <FormGroup
                      bsClass="form-group"
                      controlId="group-item-name">
                      <div
                        className="form-group">
                        <ControlLabel
                          bsClass="control-label"
                          srOnly={false}>
                          <label
                            className="control-label"
                            htmlFor="group-item-name">
                            Name
                          </label>
                        </ControlLabel>
                        <FormControl
                          bsClass="form-control"
                          componentClass="input"
                          onChange={[Function]}
                          placeholder="Enter name"
                          type="text"
                          value="Mock Organization">
                          <input
                            className="form-control"
                            id="group-item-name"
                            onChange={[Function]}
                            placeholder="Enter name"
                            type="text"
                            value="Mock Organization" />
                        </FormControl>
                      </div>
                    </FormGroup>
                  </FieldGroup>
                  <FieldGroup
                    label="Main URL"
                    name="main_url"
                    onChange={[Function]}
                    placeholder="Enter url"
                    type="text"
                    value="https://placekitten.com/">
                    <FormGroup
                      bsClass="form-group"
                      controlId="group-item-main_url">
                      <div
                        className="form-group">
                        <ControlLabel
                          bsClass="control-label"
                          srOnly={false}>
                          <label
                            className="control-label"
                            htmlFor="group-item-main_url">
                            Main URL
                          </label>
                        </ControlLabel>
                        <FormControl
                          bsClass="form-control"
                          componentClass="input"
                          onChange={[Function]}
                          placeholder="Enter url"
                          type="text"
                          value="https://placekitten.com/">
                          <input
                            className="form-control"
                            id="group-item-main_url"
                            onChange={[Function]}
                            placeholder="Enter url"
                            type="text"
                            value="https://placekitten.com/" />
                        </FormControl>
                      </div>
                    </FormGroup>
                  </FieldGroup>
                  <FieldGroup
                    label="Logo URL"
                    name="logo_url"
                    onChange={[Function]}
                    placeholder="Enter logo url"
                    type="text"
                    value="https://placekitten.com/80/80">
                    <FormGroup
                      bsClass="form-group"
                      controlId="group-item-logo_url">
                      <div
                        className="form-group">
                        <ControlLabel
                          bsClass="control-label"
                          srOnly={false}>
                          <label
                            className="control-label"
                            htmlFor="group-item-logo_url">
                            Logo URL
                          </label>
                        </ControlLabel>
                        <FormControl
                          bsClass="form-control"
                          componentClass="input"
                          onChange={[Function]}
                          placeholder="Enter logo url"
                          type="text"
                          value="https://placekitten.com/80/80">
                          <input
                            className="form-control"
                            id="group-item-logo_url"
                            onChange={[Function]}
                            placeholder="Enter logo url"
                            type="text"
                            value="https://placekitten.com/80/80" />
                        </FormControl>
                      </div>
                    </FormGroup>
                  </FieldGroup>
                  <FieldGroup
                    label="Contact"
                    name="contact"
                    onChange={[Function]}
                    placeholder="Enter contact"
                    type="text"
                    value="someone">
                    <FormGroup
                      bsClass="form-group"
                      controlId="group-item-contact">
                      <div
                        className="form-group">
                        <ControlLabel
                          bsClass="control-label"
                          srOnly={false}>
                          <label
                            className="control-label"
                            htmlFor="group-item-contact">
                            Contact
                          </label>
                        </ControlLabel>
                        <FormControl
                          bsClass="form-control"
                          componentClass="input"
                          onChange={[Function]}
                          placeholder="Enter contact"
                          type="text"
                          value="someone">
                          <input
                            className="form-control"
                            id="group-item-contact"
                            onChange={[Function]}
                            placeholder="Enter contact"
                            type="text"
                            value="someone" />
                        </FormControl>
                      </div>
                    </FormGroup>
                  </FieldGroup>
                  <FieldGroup
                    label="Email"
                    name="email"
                    onChange={[Function]}
                    placeholder="Enter email"
                    type="text"
                    value="abc@def.ghi">
                    <FormGroup
                      bsClass="form-group"
                      controlId="group-item-email">
                      <div
                        className="form-group">
                        <ControlLabel
                          bsClass="control-label"
                          srOnly={false}>
                          <label
                            className="control-label"
                            htmlFor="group-item-email">
                            Email
                          </label>
                        </ControlLabel>
                        <FormControl
                          bsClass="form-control"
                          componentClass="input"
                          onChange={[Function]}
                          placeholder="Enter email"
                          type="text"
                          value="abc@def.ghi">
                          <input
                            className="form-control"
                            id="group-item-email"
                            onChange={[Function]}
                            placeholder="Enter email"
                            type="text"
                            value="abc@def.ghi" />
                        </FormControl>
                      </div>
                    </FormGroup>
                  </FieldGroup>
                  <Button
                    active={false}
                    block={false}
                    bsClass="btn"
                    bsStyle="warning"
                    disabled={false}
                    onClick={[Function]}>
                    <button
                      className="btn btn-warning"
                      disabled={false}
                      onClick={[Function]}
                      type="button">
                      Update
                    </button>
                  </Button>
                  <Button
                    active={false}
                    block={false}
                    bsClass="btn"
                    bsStyle="danger"
                    disabled={false}
                    onClick={[Function]}>
                    <button
                      className="btn btn-danger"
                      disabled={false}
                      onClick={[Function]}
                      type="button">
                      Delete
                    </button>
                  </Button>
                </form>
              </div>
            </Col>
          </div>
        </Row>
      </div>
    </Grid>
  </EditOrganization>
</Connect(EditOrganization)>
`;

exports[`Container > EditOrganization Delete organization 1`] = `
Array [
  Object {
    "payload": "1",
    "type": "delete organization",
  },
  Object {
    "payload": Object {
      "args": Array [
        "/",
      ],
      "method": "push",
    },
    "type": "@@router/CALL_HISTORY_METHOD",
  },
]
`;

exports[`Container > EditOrganization Update organization 1`] = `
Array [
  Object {
    "payload": Object {
      "analyses": Array [],
      "analysesById": Object {},
      "contact": "My new value",
      "email": "My new value",
      "groups": Array [],
      "groupsById": Object {},
      "id": "1",
      "logo_url": "My new value",
      "main_url": "My new value",
      "name": "My new value",
      "sites": Array [],
      "sitesById": Object {},
    },
    "type": "set organization",
  },
  Object {
    "payload": Object {
      "args": Array [
        "/organizations/1",
      ],
      "method": "push",
    },
    "type": "@@router/CALL_HISTORY_METHOD",
  },
]
`;
